applicatie/static/js/custom/bulk.js

```diff
@@ -31,13 +31,108 @@ function initialiseer_form_controls_bulkcorrecties() {
 
 function bulkcorrecties_verwerken() {
 
-    let tekst = "Weet u zeker dat u de ingevoerde correcties wilt verwerken?\r\n"
-    tekst = tekst + "Zo ja,  klik op OK\r\n"
-    tekst = tekst + "Zo nee, klik op Annuleren (Cancel)"
+    let form_bulk = $("#" + FORMULIER_BULK);
+
+    let bulkcorrecties = form_bulk.find("div[id^='bulkcorr_']");
+
+    let formulier_correct_ingevuld = true;
+
+    let aantal_correcties = 0;
+
+    const _q = "\"";
+    const _a = "\{";
+    const _z = "\}";
+
+    // JSON string openen met {
+    let _bulk = _a;
+
+    $.each(bulkcorrecties, function(key, corr) {
+        /* verwerk alle bulkcorrecties in een JSON string */
+
+        let type = $(this).find("#type").children("option:selected").val();
+        let rij_soort = $(this).find("#rij_soort").children("option:selected").val();
+        let rij_code = $(this).find("#rij_code").children("option:selected").val();
+        let kolom_soort = $(this).find("#kolom_soort").children("option:selected").val();
+        let kolom_code = $(this).find("#kolom_code").children("option:selected").val();
+        let bedrag = $(this).find("#bedrag").val();
+        let omschrijving = $(this).find("#omschrijving").val();
+
+        if (type !== "" && formulier_correct_ingevuld) {
+
+            // rij_code en kolom_code moeten altijd ingevuld zijn
+            if (rij_code !== "" && kolom_code !== "") {
+
+                // dubbele quotes rondom strings zetten
+                type = _q + type + _q;
```

Comment:
Hmm dit werkt, maar is wel foutgevoelig, stel je vergeet een _q. Hier zijn toch wel standaard functies voor? bijv.:

[https://www.w3schools.com/js/js_json_stringify.asp](url)

of iets dergelijks, ik weet niet zeker of dit hetzelfde doet als je hier wilt.

Reply:
ja ik ga hier even naar kijken.
Ik maak nu gelijk een string. Misschien mooier om een object te maken en deze om te zetten naar string.

Reply:
ik heb de functie gerefactored, ik maak nu eerst een bulk object, en gebruik daarna JSON.stringify()
Ziet er volgens mij beter uit zo.

Reply:
nice!